src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/GenericDictionaryCell.cs

```diff
@@ -770,17 +770,17 @@ private bool NeedsDictionaryParameterToCallCanonicalVersion(MethodDesc method)
                 if (Method is NoMetadataMethodDesc)
                 {
                     // If the method does not have metadata, use the NameAndSignature property which should work in that case.
-                    if (!TypeLoaderEnvironment.Instance.IsStaticMethodSignature(Method.NameAndSignature.Signature))
-                        return false;
+                    if (TypeLoaderEnvironment.Instance.IsStaticMethodSignature(Method.NameAndSignature.Signature))
```

Comment:
The codepaths in this file is newly getting triggered because I'm using it for the target of the lookup (`ReadyToRunHelperId.MethodEntry`).

Eventually this will switch to the constrained call dictionary cell that .NET Native uses (that can do a proper lookup when we get `forceUseRuntimeLookup` at compile time), but that will come in a later pull request as I fix the approx constraint resolution. We can also add that to this pull request and then this change won't be needed. But it seemed like an improvement.